packages/core-web/src/styles/markbind.css

```diff
@@ -341,3 +341,14 @@ h6 {
     width: 100%;
     height: 100%;
 }
+
+.block-embed-service-powerpoint {
+    position: relative;
+    padding-bottom: 78%;
+}
+
+@media screen and (max-width: 350px) {
+    .block-embed-service-powerpoint {
+        padding-bottom: 95%;
```

Comment:
There is still some clipping (on the right side) of the slides at `320px` =\\
It also changes the aspect ratio discussed earlier here https://github.com/MarkBind/markbind/pull/1797#pullrequestreview-899134740 (I'm assuming we still want to keep this).

Consider the case the embed is even more deeply nested (say within 3 `<panel><panel><panel>`s), hardcoding the media query likely won't work as well, as it is still dependent on the parent width which has become extremely small (not the viewport's).

I really don't think there is any solution right now other from
1. fixed: easier - just means preserving the behaviour before this PR for powerpoint embeds specifically
   - you can also take some liberty in slightly reducing the values (preserving AR) to reduce overflow a little
2. minimum (harder) width
It's just physically impossible to fit even "properly designed" slides into too small and a finite number of pixels, and the powerpoint/prezi embed (unfortunately) isn't designed like the other embeds to be fully responsive. (which, would be extremely difficult given slides themselves can contain a wide range of contents)

Scrolling would be perfect, but this dosen't seem like something implementable from outside of the embed by us. (due to the embed already capturing mouse events) (again, part speculation)

Reply:
To clarify on the point of capturing mouse events for scrolling:
- Are you referring to the fact that to scroll "on" the slide itself may not work because it is being captured?

Adding a "overflow-x: scroll" in the container div like this:
![image](https://user-images.githubusercontent.com/41845017/162597991-ff98ae5d-a053-4871-9d7a-23bf5a2bcb13.png)

will give the following effect:
- cannot scroll directly on the slides
- can scroll via the scroll bar
![gif](http://g.recordit.co/wFWvj8tUEV.gif)

Reply:
Thanks for testing, yes, something like this.

Reply:
@ang-zeyu Hmm it seems like there is a slight difference from what I'm seeing locally and what the Netlify preview is showing, just for reference:

Netlify:
![image](https://user-images.githubusercontent.com/19247856/162604373-9bac1538-38e4-4e23-a79d-d947f46f1314.png)

Locally: 
![image](https://user-images.githubusercontent.com/19247856/162604371-06643eef-8c58-46cf-8e57-5ed7f6f58a48.png)

I'm not sure which one will actually be represented on the website once its actually merged, but am I correct to say that the one I see locally would be ideal?

Reply:
@ang-zeyu Sorry if I'm missing something but, I'm still not quite sure what you mean in point 2 where you set a min-width. From what I've seen, adding a min-width doesn't help at all. 

Also what do you mean by point 1? Isn't the behavior before the PR worse than hardcoding the media query. I think I can just adjust the aspect ratio (by using the padding) for the screens below 350px and that will be "good enough" for like 99% of the people.

Reply:
> @ang-zeyu Sorry if I'm missing something but, I'm still not quite sure what you mean in point 2 where you set a min-width. From what I've seen, adding a min-width doesn't help at all.

Yes, it won't work as-is (per what I'm seeing in 7194a2c).

You'll need to consider how exactly the `padding-bottom` trick works. (by setting the height as a percentage of the _parent_ width) To introduce a minimum width therefore (while preserving aspect ratio) -- you need an additional parent container. (instead of leaving it up to whatever parent container is there)

> Also what do you mean by point 1? Isn't the behavior before the PR worse than hardcoding the media query. I think I can just adjust the aspect ratio (by using the padding) for the screens below 350px and that will be "good enough" for like 99% of the people.

The solution is as discussed [here](https://github.com/MarkBind/markbind/pull/1797#issuecomment-1079230382).

There was some clipping on the right side of the slides before 7194a2c.

> "good enough" for like 99% of the people.

> Consider the case the embed is even more deeply nested (say within 3 `<panel><panel><panel>`s), hardcoding the media query likely won't work as well, as it is still dependent on the parent width which has become extremely small (not the viewport's).

Do try to understand this part as well. The issue is *not* viewport dependent.

---

In case there's still any confusion, I've been suggesting right from the first comment [here](https://github.com/MarkBind/markbind/pull/1797#discussion_r822287264) we don't sacrifice usability on mobile, and allow overflow.

The same point is also mentioned by @ryoarmanda here https://github.com/MarkBind/markbind/pull/1797#issuecomment-1079230382.

I feel really strongly about this for a couple of reasons:
- It seems really strange that a content-oriented static site generator for education websites would compromise on readability on specific devices
- The external onedrive link (bottom right button in the embed) is even less usable on mobile than our previous behaviour before this PR. Were there a reasonable alternative (such as youtube's fullscreen mode), this would be different.
- > Not covering our use cases. Overflow is still possible if there is a page with PowerPoint embeds. (but low impact as its not highly used)

   As mentioned by @ryoarmanda here as well, overflow is of very low impact since powerpoint embeds are seldom used. In which case, I feel the case for usability is stronger.

... This is just my opinion. If there is consensus otherwise (which I have been repeatedly trying to get confirmation on thus far), I am *amenable*. (do post an update here as I am out of the loop)

Reply:
Hi @ang-zeyu, I think @ong6 and I had quite a few discussions over this PR previously, about how to create a good enough solution. However, we were really getting confused with the whole conversation (which we cross-reference earlier comments here and there 🤔) within this PR thread and some of the suggestions that we tried didn't really work out (e.g maybe works for Youtube and Box, but not for slides) as expected or as suggested.

Wonder if you can give any consolidated advice on how to proceed and what are the specific deliverables that we can aim to achieve for now and what can be set as a future TODO for a separate PR? Thanks!

Reply:
I think something like this:
- #1741 - this has been fixed as a "freebie" with the sticky header PR. This was just a point that a huge part of the overflow problem you see here is unrelated to embeds specifically (e.g. fixed-size images). Hence why I'm really not concerned with overflow as much here too. (see latest netlify [preview](https://markbind-master.netlify.app/userguide/formattingcontents#embeds))
- 2 possible solutions we can do right now:
   - just keep the existing behaviour, styling specifically for ppt embeds, scale the rest
   - Keep the best of both worlds. That is, while *keeping the aspect ratio*, scale the ppt embed after some minimum width threshold (we can just use the existing fixed width as one), but otherwise, the embed should minimally be kept at that width. Hint in the first 2 paragraphs in this [comment](https://github.com/MarkBind/markbind/pull/1797#discussion_r846779449). I mentioned this is harder, but not much. Its a fairly interesting exercise in css and I really encourage trying it out.
- Future: originally, something along the lines of exploring the embed scrolling idea further. But with the sticky header PR's UX change, this is even less necessary, although that makes its own [compromises](https://github.com/MarkBind/markbind/pull/1939#discussion_r895126866). Wherever possible, we can aim to scale things to 100% container width (i.e. not something like the ppt embed that scales poorly, but for example images, or in the case of this PR youtube embeds), so page overflow becomes even more seldom. But this is really up for discussion individually.

Fwiw, I found this excerpt from mdn some time ago. I think it helps explain my view on usabililty v "looking nicer" with much better generality, which might have been more helpful:

> The problem with data loss is that you might not notice. Website visitors may not notice.
> 
> \- https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Overflowing_content#css_tries_to_avoid_data_loss

Reply:
> * just keep the existing behaviour, styling specifically for ppt embeds, scale the rest

Hi @ang-zeyu thanks for giving the advice, I think I'll just go with this simpler approach and make another issue inviting people to take up the CSS challenge. Will try to push one PR tmr to clean up and make final changes.

Reply:
Thanks for updating this!

> just keep the existing behaviour, styling specifically for ppt embeds, scale the rest

Existing behaviour meaning keeping the fixed width and height before this PR, I can't see any difference right now with the ppt embed from the other ones (let me know if I'm missing something!). Which is in line with the rest of the point on not compromising usability / data loss.

One simple way to do it is using the existing `block-embed-service-powerpoint` class to target ppt embeds only.

Some other nits below:

Reply:
Ok i have reverted to the original, there is no easy way to do this without completely removing the block-embed css style from the ppt embed.

Reply:
> there way no easy way

This is looks simple enough too. But fwiw I was thinking of these 2 ways off the top of my head, the second being the simplest:
- you can rely on the css cascade to override the `block-embed` styles, e.g.
   ```css
  .block-embed-powerpoint {
      overflow-x: visible;
      position: static;
      width: auto;
  }
   ```
- That's a little verbose; Alternatively you could simply add a [` :not`](https://developer.mozilla.org/en-US/docs/Web/CSS/:not)
   ```css
  .block-embed:not(.block-embed-powerpoint) {
      overflow-x: auto;
      position: relative;
      width: 100%;
  }
   ```

Reply:
ahh thanks for the info, i really need to work on my css 😄